src/nlp/arrow_dataset.py

```diff
@@ -1075,3 +1190,54 @@ def train_test_split(
         )
 
         return {"train": train_split, "test": test_split}
+
+    def add_vector_index(
```

Comment:
I think I would call it `add_faiss_index`.

In general, we may want to be compatible with a diversity of ann-engines. We'll probably have a way to register ann engine at some point and call a `dataset.add_index(registered_engine, **kwargs)`. But probably for now we can just have a simple API focusing on FAISS and ES and leave this for future work after we've had some feedback from the community.